src/NHibernate/Type/AbstractTimeOnlyType.cs

```diff
@@ -0,0 +1,134 @@
+#if NET6_0_OR_GREATER
+using System;
+using System.Data.Common;
+using System.Globalization;
+using NHibernate.Engine;
+using NHibernate.SqlTypes;
+
+namespace NHibernate.Type
+{
+	/// <summary>
+	/// Base class for TimeOnly types.
+	/// Fractional seconds precision:
+	///   null  => preserve full precision (no truncation)
+	///   0..6  => truncate (floor) to that many fractional digits
+	///   >=7   => preserve full precision
+	/// </summary>
+	[Serializable]
+	public abstract class AbstractTimeOnlyType<TParameter> : PrimitiveType
+	{
+		private static readonly int[] TicksPerPrecision = { 10_000_000, 1_000_000, 100_000, 10_000, 1_000, 100, 10 };
+		private readonly int _ticksForPrecision;
+		private readonly bool _truncate;
+		private readonly byte? _fractionalSecondsPrecision;
+
+		protected AbstractTimeOnlyType(byte? fractionalSecondsPrecision, SqlType sqlType) : base(sqlType)
+		{
+			_fractionalSecondsPrecision = fractionalSecondsPrecision;
+			if (fractionalSecondsPrecision is >= 0 and <= 6)
+			{
+				_ticksForPrecision = TicksPerPrecision[fractionalSecondsPrecision.Value];
+				_truncate = true;
+			}
+			else
+			{
+				// null or >=7 => full precision, no truncation
+				_ticksForPrecision = 1;
+				_truncate = false;
+			}
+		}
+
+
+		public override string Name => GetType().Name.EndsWith("Type", StringComparison.Ordinal)
```

Comment:
Nit: undue blank line.


```suggestion
		}

		public override string Name => GetType().Name.EndsWith("Type", StringComparison.Ordinal)
```